.github/workflows/pkgci_test_amd_mi355.yml

```
@@ -0,0 +1,102 @@
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  test_amd_mi355:
+    name: "test_mi355"
+    strategy:
+      fail-fast: false
+    runs-on: linux-mi35x-1gpu-ossci-iree-org
+    timeout-minutes: 60
+    if: github.event_name != 'pull_request' || github.event.pull_request.draft == false
+    env:
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      BUILD_DIR: build-tests
+      VENV_DIR: ${{ github.workspace }}/.venv
+      GH_TOKEN: ${{ github.token }}
+      IREE_CPU_DISABLE: 1
+      IREE_VULKAN_DISABLE: 1
+      IREE_CUDA_ENABLE: 0
+      IREE_HIP_ENABLE: 1
+      IREE_HIP_TEST_TARGET_CHIP: "gfx950"
+    container:
+      image: 'ghcr.io/rocm/no_rocm_image_ubuntu24_04@sha256:405945a40deaff9db90b9839c0f41d4cba4a383c1a7459b28627047bf6302a26'
+      options: --ipc host
+        --group-add video
+        --device /dev/kfd
+        --device /dev/dri
+        --env-file /etc/podinfo/gha-gpu-isolation-settings
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: "Setting up Python"
+        id: setup_python
+        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
+        with:
+          python-version: "3.11"
+      - name: Create Python venv
+        run: |
+           python -m venv ${VENV_DIR}
+           source ${VENV_DIR}/bin/activate
+      - name: Check out repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          submodules: false
+      - name: Check out runtime submodules
+        run: ./build_tools/scripts/git/update_runtime_submodules.sh
+      - name: Install dependencies
+        if: ${{ inputs.artifact_run_id != '' }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install gh
+          # Install TheRock
+          sudo apt install -y gfortran build-essential binutils
+          python -m pip install \
+            --index-url https://d2awnip2yjpvqn.cloudfront.net/v2/gfx950-dcgpu/ \
+            rocm[libraries,devel]
+
+          # Set environment variables
+          export ROCM_PATH=$(python -m rocm_sdk path --root)
+          export LD_LIBRARY_PATH="$ROCM_PATH/lib":$LD_LIBRARY_PATH
+          echo "ROCM_PATH=$ROCM_PATH" >> $GITHUB_ENV
+          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH" >> $GITHUB_ENV
+      - uses: actions/download-artifact@634f93cb2916e3fdff6788551b99b062d0335ce0 # v5.0.0
+        if: ${{ inputs.artifact_run_id == '' }}
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+      - name: Setup base venv
+        run: |
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+      - name: Build tests
+        run: ./build_tools/pkgci/build_tests_using_package.sh ${VENV_DIR}/bin
+      - name: Run GPU tests
```

I'd suggest putting a line break here to differentiate between setup code and actual test code, matching the other workflows

```suggestion
      - name: Setup base venv
        run: |
          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
            --fetch-gh-workflow=${{ inputs.artifact_run_id }}

      - name: Build tests
        run: ./build_tools/pkgci/build_tests_using_package.sh ${VENV_DIR}/bin
      - name: Run GPU tests
```